fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocks.g4

```diff
@@ -355,21 +355,39 @@ intervalField
     ;
 
 type
-    : arrayType
-    | baseType ('(' typeParameter (',' typeParameter)* ')')?
+    : baseType
     | decimalType ('(' precision=typeParameter (',' scale=typeParameter)? ')')?
+    | arrayType
     ;
 
 arrayType
     : ARRAY '<' type '>'
     ;
 
 typeParameter
-    : INTEGER_VALUE | type
+    : INTEGER_VALUE
     ;
 
 baseType
-    : identifier
+    : BOOLEAN
+    | TINYINT
+    | SMALLINT
+    | INT
+    | INTEGER
```

Comment:
MySQL allows `int(11)` like syntax, do we need to be compatible with it?

Reply:
Yes, but this type is currently not supported in queries, only in create table. We can put this in the support of create table PR